idea-plugin/src/main/kotlin/arrow/meta/plugin/idea/plugins/higherkinds/HigherKindsIdePlugin.kt

```diff
@@ -5,16 +5,20 @@ import arrow.meta.invoke
 import arrow.meta.plugin.idea.IdeMetaPlugin
 import arrow.meta.plugin.idea.resources.ArrowIcons
 import arrow.meta.plugins.higherkind.isHigherKindedType
+import org.jetbrains.kotlin.lexer.KtTokens
 import org.jetbrains.kotlin.psi.KtClass
 import org.jetbrains.kotlin.utils.addToStdlib.safeAs
 
 val IdeMetaPlugin.higherKindsIdePlugin: Plugin
   get() = "HigherKindsIdePlugin" {
     meta(
-      addLineMarkerProvider(
+      addFastLineMarkerProvider(
         icon = ArrowIcons.HKT,
         message = "HKT",
-        matchOn = { it.safeAs<KtClass>()?.let(::isHigherKindedType) == true }
+        matchOn = {
+          KtTokens.IDENTIFIER == it.node?.elementType
```

Comment:
@jansorg Why was the parent-child check necessary?

Reply:
see https://github.com/JetBrains/intellij-community/blob/192/platform/lang-api/src/com/intellij/codeInsight/daemon/LineMarkerProvider.java#L22. 
We're making sure that we're only adding the marker for the class name identifier, not the whole `KtClass`.
A `KtClass` covers all of this `@higherkind class Name {...}`, the identifier leaf element is just `Name` here. See the javadoc above for a few more details on this.

Reply:
What does difference imply for the user? I understand your point but I am curious to understand the use-case being an IDEA user.

Reply:
AFAIK they'll be situations, where the Icons blink annoyingly.
I am sure we can find workarounds to prevent that, but we don't have any User data.
In this PR we discuss it more in-depth https://github.com/arrow-kt/arrow-meta/pull/12